server/image_pull.go

```diff
@@ -17,7 +21,32 @@ func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (*pb.P
 	if img != nil {
 		image = img.Image
 	}
+	var (
+		username string
+		password string
+	)
+	if req.GetAuth() != nil {
+		username = req.GetAuth().Username
+		password = req.GetAuth().Password
+		if req.GetAuth().Auth != "" {
+			var err error
+			username, password, err = decodeDockerAuth(req.GetAuth().Auth)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
 	options := &copy.Options{}
+	// a not empty password should be sufficient to decide whether to send auth
+	// or not I guess
+	if password != "" {
```

Comment:
Although I can see people using password less accounts, I really don't know how they could have username less ones. So I'd use `username` not being empty as a criteria to send `auth`, instead of `password`.